packages/query-core/src/__tests__/hydration.test.tsx

```
@@ -1385,4 +1385,177 @@ describe('dehydration and rehydration', () => {
     // error and test will fail
     await originalPromise
   })
+
+  // Companion to the test above: when the query already exists in the cache
+  // (e.g. after an initial render or a first hydration pass), the same
+  // synchronous thenable resolution must also produce status: 'success'.
+  // Previously the if (query) branch would spread status: 'pending' from the
+  // server state without correcting it for the resolved data.
+  test('should set status to success when rehydrating an existing pending query with a synchronously resolved promise', async () => {
```

ESLint

[error] Prefer using it instead of test within describe

(vitest/consistent-test-it)